lua/rustaceanvim/config/init.lua

```diff
@@ -117,6 +118,22 @@ vim.g.rustaceanvim = vim.g.rustaceanvim
 ---@alias dap_adapter_type_executable "executable"
 ---@alias dap_adapter_type_server "server"
 
+---@class DapClientConfig
+---@field type string The dap adapter to use
+---@field name string
+---@field request dap_config_request_launch | dap_config_request_attach | dap_config_request_custom The type of dap session
+---@field cwd? string Current working directory
+---@field program? string Path to executable for most DAP clients
+---@field args? string[] Optional args to DAP client, not valid for all client types
+---@field env? string  Environmental variables
+---@field initCommands? string Initial commands to run, `lldb` clients only
+---@field sourceMap? string Map of source files, `lldb` clients only
+---@field coreConfigs? table Essential config values for `probe-rs` client, see https://probe.rs/docs/tools/debugger/
```

Comment:
The `probe-rs` site has a bunch of exmaples:
https://probe.rs/docs/tools/debugger/#start-a-debug-session-with-minimum-configuration
Otherwise here is a basic config I am currently using:
https://gist.github.com/Andrew-Collins/1f8ce17ec1372e9f5a24755a3eed89a5

Reply:
I can easily create an example project for you if you like, but that would require having compatible hardware. So let me know what you have access to.

Reply:
No need. The `probe-rs` examples should be enough.
I am hoping to add integration tests for all of the dap features eventually. But if it's dependent on hardware, that might not be a good idea for `probe-rs`.